diff --git a/configure.ac b/configure.ac
index 970a2141b..9614de01f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -151,7 +151,7 @@ if test "x$bindir" = "x\${exec_prefix}/bin"; then
 fi
 
 if test "x$with_no_install" != "x"; then
-	configpath="${PWD}/lib/"
+	configpath="./lib/"
 else
 	configpath="${sysconfdir}/${PACKAGE}/"
 fi
@@ -163,7 +163,7 @@ esac
 
 
 if test "x$with_no_install" != "x"; then
-	libpath="${PWD}/lib/"
+	libpath="./lib/"
 	bindir=".."
 else
 	libpath="${datarootdir}/${PACKAGE}/"
@@ -175,7 +175,7 @@ case "/$libpath" in
 esac
 
 if test "x$with_no_install" != "x"; then
-	docdir="${PWD}/doc/"
+	docdir="./doc/"
 else
 	# Only change docdir if it's the configure-supplied default, which handily doesn't expand prefix
 	if test "x$docdir" = "x\${datarootdir}/doc/\${PACKAGE_TARNAME}"; then
@@ -189,7 +189,7 @@ case "/$docdir" in
 esac
 
 if test "x$with_no_install" != "x"; then
-	varpath="${PWD}/lib/"
+	varpath="./lib/"
 else
 	varpath="${localstatedir}/games/${PACKAGE}/"
 fi
diff --git a/src/init.c b/src/init.c
index 0669c5988..dc58e312b 100644
--- a/src/init.c
+++ b/src/init.c
@@ -376,19 +376,17 @@ void init_file_paths(const char *configpath, const char *libpath, const char *da
 
 #endif /* PRIVATE_USER_PATH */
 
+#ifndef USE_PRIVATE_PATHS
+	BUILD_DIRECTORY_PATH(ANGBAND_DIR_USER, datapath, "user");
+#endif /* USE_PRIVATE_PATHS */
+
 	/* Build the path to the user info directory */
 	BUILD_DIRECTORY_PATH(ANGBAND_DIR_INFO, ANGBAND_DIR_USER, "info");
 
-#ifdef USE_PRIVATE_PATHS
-	userpath = ANGBAND_DIR_USER;
-#else /* !USE_PRIVATE_PATHS */
-	userpath = (char *)datapath;
-#endif /* USE_PRIVATE_PATHS */
-
 	/* Build the path to the score, save and archive directories */
-	BUILD_DIRECTORY_PATH(ANGBAND_DIR_SCORES, userpath, "scores");
-	BUILD_DIRECTORY_PATH(ANGBAND_DIR_SAVE, userpath, "save");
-	BUILD_DIRECTORY_PATH(ANGBAND_DIR_ARCHIVE, userpath, "archive");
+	BUILD_DIRECTORY_PATH(ANGBAND_DIR_SCORES, ANGBAND_DIR_USER, "scores");
+	BUILD_DIRECTORY_PATH(ANGBAND_DIR_SAVE, ANGBAND_DIR_USER, "save");
+	BUILD_DIRECTORY_PATH(ANGBAND_DIR_ARCHIVE, ANGBAND_DIR_USER, "archive");
 
 #undef BUILD_DIRECTORY_PATH
 }
diff --git a/src/tests/Makefile b/src/tests/Makefile
index ed42b69c5..73a89c7c4 100644
--- a/src/tests/Makefile
+++ b/src/tests/Makefile
@@ -18,7 +18,7 @@ TESTOBJS += test-utils.o unit-test.o
 build : $(TESTPROGS)
 
 run : build
-	@./run-tests
+	@(cd ../.. ; ./src/tests/run-tests)
 
 %.o : %.c
 	@$(CC) $(CFLAGS) $(CPPFLAGS) -c -o $@ $^
